### PR TITLE
Enable rabbitmq to track finished function activations

### PR DESCRIPTION
**Pull request by JosepSampe**
By using this service: https://console.bluemix.net/catalog/services/cloudamqp, it is possible to quickly track the functions that have finished.

Add the AMQP_URL in the config file:
```
rabbitmq: 
     amqp_url : <AMQP_URL>
```
This feature is integrated in the `pw.monitor()` method.
To activate it, use: `pw = pywren.ibm_cf_executor(use_rabbitmq=True)`

If the rabbitmq config is not in the config file, the feature is ignored by PyWren, even if `use_rabbitmq=True`

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by gilv** [comment]
@JosepSampe pika need to be added to PyWren dependecies, otherwise exception. 
Can you please add it?

gilv-mac:runtime gilv$ ./deploy_runtime create cactusone/pywren-dev:3.6
Traceback (most recent call last):
  File "./deploy_runtime", line 5, in <module>
    import pywren_ibm_cloud as pywren
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pywren_ibm_cloud/__init__.py", line 17, in <module>
    from pywren_ibm_cloud.wren import *
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pywren_ibm_cloud/wren.py", line 28, in <module>
    from pywren_ibm_cloud.executor import Executor
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pywren_ibm_cloud/executor.py", line 26, in <module>
    from pywren_ibm_cloud.wait import wait
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pywren_ibm_cloud/wait.py", line 18, in <module>
    import pika
ModuleNotFoundError: No module named 'pika'

**Comment by gilv** [comment]
@JosepSampe do you manage to run tests - python3 test/testpywren.py ?
I created new run time based on this branch, and tests fails there.

**Comment by gilv** [comment]
@JosepSampe map_reduce fails

2019-02-11T10:43:36.928776722Z stdout: ------------------- FUNCTION LOG -------------------
2019-02-11T10:43:36.928784403Z stdout: ----------------------------------------------------
2019-02-11T10:43:36.928789207Z stdout: [INFO] jobrunner: Function: Success execution
2019-02-11T10:43:37.86633455Z  stdout: [DEBUG] pywren_ibm_cloud.storage.backends.cos: PUT Object Apywren.jobs/d0d661a6-8a68/cd876c30/00003/output.pickle - Size: 38.0B - OK
2019-02-11T10:43:37.866392625Z stdout: [INFO] jobrunner: Jobrunner finished
2019-02-11T10:43:37.997970578Z stdout: [INFO] wrenhandler: Command execution finished
2019-02-11T10:43:37.999952204Z stderr: Traceback (most recent call last):
2019-02-11T10:43:38.000083899Z stderr: File "pythonrunner.py", line 88, in run
2019-02-11T10:43:38.000099845Z stderr: exec('fun = %s(param)' % self.mainFn, self.global_context)
2019-02-11T10:43:38.000188544Z stderr: File "<string>", line 1, in <module>
2019-02-11T10:43:38.00035109Z  stderr: File "__main__.py", line 27, in main
2019-02-11T10:43:38.000368986Z stderr: ibm_cloud_function_handler(args)
2019-02-11T10:43:38.00037642Z  stderr: File "/action/pywren_ibm_cloud/action/wrenhandler.py", line 209, in ibm_cloud_function_handler
2019-02-11T10:43:38.00038376Z  stderr: rabbit_amqp_url = config['rabbitmq'].get('amqp_url')
2019-02-11T10:43:38.000518607Z stderr: KeyError: 'rabbitmq'

**Comment by JosepSampe** [comment]
In my case all ran successfully. That error seems to be due to the runtime is not updated with new changes. Did you run `./deploy_runtime update runtime/name`? Or if you created new runtime with this branch, did you update the config file to point to this as default runtime?

**Comment by JosepSampe** [comment]
@gilv Yep, "create" should work anyway

**Comment by JosepSampe** [comment]
@gilv Yep, "create" should work anyway. 

In the PyWren config dictionary, the 'rabbitmq' key is always created by default, even if you don't set it in the config file: https://github.com/pywren/pywren-ibm-cloud/pull/84/files#diff-fa8dc87e510a994c3625d8e7669efa2aR148

**Comment by gilv** [comment]
@JosepSampe it works now :)

**Comment by gilv** [comment]
@JosepSampe very small comment, can you please add log.info message to the init indicating status of rabbitmq---activated or disabled? Thanks

File: pywren/pywren_ibm_cloud/wren.py [hunk 1]
```diff
@@ -79,6 +79,8 @@ def __init__(self, config=None, runtime=None, log_level=None, use_rabbitmq=False
 
         if not use_rabbitmq:
             self.config['rabbitmq']['amqp_url'] = None
+        else:
+            logger.debug('Going to use RabbitMQ to track function activations')
```
Comment (gilv) [inline pywren/pywren_ibm_cloud/wren.py]: @JosepSampe it's log.debug ()...not log.info() ;)

**Comment by gilv** [comment]
@JosepSampe this is great addition..thanks

**Pull request merged**